spark/src/main/scala/org/apache/spark/sql/delta/DeltaOperations.scala

```diff
@@ -473,8 +480,9 @@ object DeltaOperations {
     override val operationMetrics: Set[String] = DeltaOperationMetrics.RESTORE
   }
 
-  sealed abstract class OptimizeOrReorg(override val name: String, predicates: Seq[Expression])
-    extends OperationWithPredicates(name, predicates)
+  sealed abstract class OptimizeOrReorg(override val name: String, predicates: Seq[Expression]
+                                        , metadata: Metadata)
```

Comment:
```suggestion
                                        metadata: Metadata)
```
I like this style personally, but it's not common in the Delta codebase